File: Miwok/app/src/main/java/com/example/miwok/WordAdapter.java
package com.example.miwok;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.content.ContextCompat;

import java.util.ArrayList;

public class WordAdapter extends ArrayAdapter<Word> {

    private int mColorResourceId;

    private static final String LOG_TAG = WordAdapter.class.getSimpleName();

    public WordAdapter(Activity context , ArrayList<Word> words){
        super(context , 0 , words);
    }

    public View getView(int position , View convertView , ViewGroup parent){
        View listItemView = convertView;

        if(listItemView == null){
            listItemView = LayoutInflater.from(getContext()).inflate(R.layout.list_item , parent , false);
        }

        Word currentWord = getItem(position);

        TextView defaultTranslation = (TextView) listItemView.findViewById(R.id.default_text_view);
        defaultTranslation.setText(currentWord.getDefaultTranslation());

        TextView miwokTranslation = (TextView) listItemView.findViewById(R.id.miwok_text_view);
        miwokTranslation.setText(currentWord.getMiwokTranslation());

        ImageView imageView = (ImageView) listItemView.findViewById(R.id.image);
        if(currentWord.hasImage()){
            imageView.setImageResource(currentWord.getmImageResourceId());
            imageView.setVisibility(View.VISIBLE);
        }else{
            imageView.setVisibility(View.GONE);
        }

        /*
        View textContainer = listItemView.findViewById(R.id.text_container);

        int color = ContextCompat.getColor(getContext() , mColorResourceId);

        textContainer.setBackgroundColor(color);

        */

        return listItemView;

    }

}

File: AEWRoster/app/src/main/java/com/example/aewroster/WordAdapter.kt
package com.example.aewroster

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.lifecycle.AndroidViewModel
import androidx.recyclerview.widget.RecyclerView

class WordAdapter(context: Context , private val listener : IRVWordAdapter) : RecyclerView.Adapter<WordAdapter.WordViewHolder>() {

   private val allWords = ArrayList<Word>()

    inner class WordViewHolder(itemView : View) : RecyclerView.ViewHolder(itemView){
        val name = itemView.findViewById<TextView>(R.id.name)
        val delete = itemView.findViewById<ImageView>(R.id.delete)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WordViewHolder {
        val viewHolder = WordViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.word_item , parent , false))
        viewHolder.delete.setOnClickListener{
            listener.onDelete(allWords[viewHolder.adapterPosition])
        }
        return viewHolder
    }

    override fun getItemCount(): Int {
       return allWords.size
    }

    override fun onBindViewHolder(holder: WordViewHolder, position: Int) {
        val currentWord = allWords[position]
        holder.name.text = currentWord.name
    }

    fun update(newList : List<Word>){
        allWords.clear()
        allWords.addAll(newList)
        notifyDataSetChanged()
    }

}

interface IRVWordAdapter{
    fun onDelete(word : Word)
}